src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -82,4 +80,73 @@ public void updateMeal(int mealId, Meal meal) {
         System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
     }
 
-}
\ No newline at end of file
+
+    @Override
+    public Map<String, List<Meal>> mealsByCategories() {
+        List<Meal> allMeals = getMeals();
+        Map<String, List<Meal>> mealsByCategoriesList = new HashMap<>();
+        for (Meal meal : allMeals) {
+            mealsByCategoriesList.computeIfAbsent(meal.getCategory(), k -> new ArrayList<>()).add(meal);
+        }
+
+        return mealsByCategoriesList;
+    }
+
+    private int randomNumberGenerator(int max) {
+        Random random = new Random();
+        return random.nextInt(max);
+    }
+
+
+    @Override
+    // generate 7 random meals per each day
+    public HashMap<DaysOfTheWeek, Meal> generateWeeklyMeals() {
+
+        HashMap<DaysOfTheWeek, Meal> weeklyPlan = new HashMap<>();
+        Map<String, List<Meal>> mealsByCategories = mealsByCategories();
+
+        // create a method ↓
+        HashMap<String, Integer> request = new HashMap<>();
+        request.put("Chicken", 2);
+        request.put("Fish", 1);
+        request.put("Meat", 2);
+        request.put("Vegetarian", 2);
```

Comment:
Create a separate method and call it once this method is called, ie requested body user input

Reply:
add a check if user does not provide an input, use a build in